hub/core/storage/ipfs.py

```diff
@@ -0,0 +1,483 @@
+from aifc import Error
+from distutils.log import error
+import hub
+import time
+import json
+import logging
+import dag_cbor
+import requests
+import pandas as pd
+
+from io import StringIO,BytesIO
+from requests.exceptions import HTTPError
+from typing import Optional, Set, Sequence
+from hub.core.storage.provider import StorageProvider
+from typing import Union, List
+
+from ipfshttpclient.multipart import stream_files, stream_directory
+
+
+logger = logging.getLogger("ipfsspec")
+
+class IPFSProvider(StorageProvider):
+    def __init__(
+        self,
+        coreurl:str=None, # Core URL to use
+        cid:str='',
+        storage_type:str=None, # specify type of gateway (e.g. Infura, Estuary, Web3.Storage, local node...)
```

Comment:
please use proper docstrings similar to s3 and gcs provider instead of inline comments